src/python/pants/backend/tools/trufflehog/rules.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+""""Generates trufflehog rules."""
+
+from __future__ import annotations
+
+from pants.core.goals.lint import LintFilesRequest, LintResult
+from pants.core.util_rules.config_files import ConfigFiles, ConfigFilesRequest
+from pants.core.util_rules.external_tool import DownloadedExternalTool, ExternalToolRequest
+from pants.core.util_rules.partitions import Partitions
+from pants.engine.fs import (
+    CreateDigest,
+    Digest,
+    DigestEntries,
+    FileEntry,
+    MergeDigests,
+    PathGlobs,
+    Snapshot,
+)
+from pants.engine.platform import Platform
+from pants.engine.process import FallibleProcessResult, Process
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.source.filespec import FilespecMatcher
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+from .subsystem import Trufflehog
+
+
+class TrufflehogRequest(LintFilesRequest):
+    """Defines the request for linting, in support of batching files in chunks, in case many files
+    need to be scanned)"""
+
+    tool_subsystem = Trufflehog
+
+
+@rule
+async def partition_inputs(
+    request: TrufflehogRequest.PartitionRequest, trufflehog: Trufflehog
+) -> Partitions:
+    """Configure the partitions scheme."""
+    if trufflehog.skip or not request.files:
+        return Partitions()
+
+    matched_filepaths = FilespecMatcher(
+        includes=["**"],
+        excludes=trufflehog.exclude_options,
+    ).matches(tuple(request.files))
+    return Partitions.single_partition(sorted(matched_filepaths))
+
+
+@rule(desc="Run Trufflehog Scan", level=LogLevel.DEBUG)
+async def run_trufflehog(
+    request: TrufflehogRequest.Batch,
+    trufflehog: Trufflehog,
+    platform: Platform,
+) -> LintResult:
+    """Runs the trufflehog executable against the targeted files."""
+
+    download_trufflehog_get = Get(
+        DownloadedExternalTool, ExternalToolRequest, trufflehog.get_request(platform)
+    )
+
+    config_files_get = Get(ConfigFiles, ConfigFilesRequest, trufflehog.config_request())
+
+    downloaded_trufflehog, config_digest = await MultiGet(download_trufflehog_get, config_files_get)
+    # the downloaded files are going to contain the `exe`, readme and license. We only
+    # want the `exe`
+    entry = next(
+        e
+        for e in await Get(DigestEntries, Digest, downloaded_trufflehog.digest)
+        if isinstance(e, FileEntry) and e.path.endswith("trufflehog")
```

Comment:
It probably doesn't matter, but this is very slightly fragile in case there ever is some other file in there ending in `trufflehog`, like, IDK, `setup_trufflehog` or something. Maybe check os.path.basename?

Reply:
Fair, I updated this so that we are checking that **exact** path, which turns out to be `trufflehog`. I also added a check to only load executable files.